src/Tools/ExternalAccess/Extensions/External/DocumentLinePosition.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text.Json.Serialization;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Extensions;
+
+/// <summary>
+/// Immutable representation of a line number and position within a document.
+/// </summary>
+public readonly struct DocumentLinePosition
+{
+    /// <summary>
+    /// Initializes a new instance of a <see cref="DocumentLinePosition"/> with the given document and line position.
+    /// </summary>
+    /// <param name="document">The document.</param>
+    /// <param name="linePosition">The position within the document</param>
+    /// <exception cref="ArgumentNullException">When <paramref name="document"/> is <see langword="null"/>.</exception>
+    /// <exception cref="InvalidOperationException">When the document path is not available.</exception>
+    public DocumentLinePosition(Document document, LinePosition linePosition)
```

Comment:
seems odd.  when is roslyn making instnaces of these?  i woudl not have expected referencing any roslyn specific types here (since they won't be in VS when this type eventually moves there).

Reply:
Roslyn is never instantiating any of these values (except for when deserializing json if the extender included some of these in the message type).
This constructor, while not strictly necessary, is meant to allow the extender, when writing their handler, to easily create these objects. The handler is meant to have a dependency on `Microsoft.CodeAnalysis.Workspaces`.

Reply:
gotcha.